tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -146,7 +146,13 @@ public static void initTrace(Observability observability, String traceId) {
      */
     @CheckReturnValue
     public static OpenSpan startSpan(String operation, String parentSpanId, SpanType type) {
-        return getOrCreateCurrentTrace().startSpan(operation, parentSpanId, type);
+        Trace trace = getOrCreateCurrentTrace();
+        boolean wasEmpty = trace.isEmpty();
```

Comment:
This is kinda sad - but definitely don't want to be setting the MDC property on every span open, so we have to know if it could have become active here. If traces had to be initialized with a span, this would be part of `init/setTrace` :(